pandas/tests/indexes/interval/test_setops.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import pytest
+
+from pandas import Index, IntervalIndex, Timestamp, interval_range
+import pandas.util.testing as tm
+
+
+@pytest.fixture(scope='class', params=[None, 'foo'])
+def name(request):
+    return request.param
+
+
+class TestIntervalIndex:
```

Comment:
I know you just copied and pasted, but I think is a great opportunity to move from an OOP class and go functional instead per `pytest` idiom.

Reply:
@gfyoung Would you give me more guidelines or examples? I don't know what you want.

Reply:
Sure thing: `pytest` prefers that you define test functions directly instead test classes (less overhead).  Notice that in a lot of your tests, you don't really use the `self` parameter.  And here's why: you have `create_index` as an instance method, but it's really a static method.

So the first thing you do is move `create_index` outside and drop the `self` parameter:

~~~python
def create_index(closed="right"):
    return IntervalIndex.from_breaks(range(11), closed=closed)
~~~

Now let's take a look at `test_union`.  That's a massive test!  There's actually about five tests in that one test alone.  But we'll get to that step later on (separate comment).  Let's focus first making this test "functional".  Essentially, what that means is moving the test out of your class, and dropping the `self` parameter from the definition.

~~~python
@pytest.mark.parametrize(...)
def test_union(closed, sort):
    index = create_index(closed=closed)
    ...   # rest of test as before
~~~

This is a functional test because it's just a test function, no separate wrapping by a class definition (it doesn't have a `self` parameter).

BTW, for more info on `pytest` to get a better for how it works:

https://docs.pytest.org/en/latest/
https://docs.pytest.org/en/latest/example/

Reply:
@gfyoung Thanks for your detail reply.